### PR TITLE
Add `cosine_decay_restarts` to available LR schedules.

### PR DESCRIPTION
**Pull request by virajbshah**
Adds the `tf.train.cosine_decay_restarts` learning rate schedule.

### CODE REVIEW

**Review by ondrasej — APPROVED** [review]
Nice!
